### PR TITLE
test: apps

### PR DESCRIPTION
**Pull request by cachho**
## Description

I was surprised to find out, that we don't have any basic tests for initializing the app. This should be the first thing to test and is not very expensive.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] test

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [X] Unit Test

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by cachho** [comment]
Fails because sentence transformers is not installed. It works in local dev, because sentence transformers is a poetry requirement, but not part of setup.py. This has to be streamlined.

File: tests/apps/test_app.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import unittest
+
+from embedchain import App
+from embedchain.config import AppConfig
+
+
+class TestChromaDbHosts(unittest.TestCase):
```
Comment (taranjeet) [inline tests/apps/test_app.py:7]: TestApp

File: tests/apps/test_custom_app.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from embedchain import CustomApp
+from embedchain.config import CustomAppConfig
+from embedchain.models import EmbeddingFunctions, Providers
+
+
+class TestChromaDbHosts(unittest.TestCase):
```
Comment (taranjeet) [inline tests/apps/test_custom_app.py:8]: TestCustomApp

File: tests/apps/test_open_source_app.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import unittest
+
+from embedchain import OpenSourceApp
+from embedchain.config import OpenSourceAppConfig
+
+
+class TestChromaDbHosts(unittest.TestCase):
```
Comment (taranjeet) [inline tests/apps/test_open_source_app.py:7]: TestOpenSourceApp

**Comment by taranjeet** [comment]
Hi @cachho , left comments around class name.
Also why are the tests failing?

**Comment by cachho** [comment]
> Hi @cachho , left comments around class name. Also why are the tests failing?

> Fails because sentence transformers is not installed. It works in local dev, because sentence transformers is a poetry requirement, but not part of setup.py. This has to be streamlined.

With that it's easy to see that we have 0 tests for alternative app types.

**Comment by cachho** [comment]
> can you figure out what needs to be done for "sentence-transformers"?

opened PR #416 for this.

**Comment by cachho** [comment]
> Also name change comment still pending.

done

**Comment by cachho** [comment]
I guess this change doesn't make sense until we have figured out #416

**Comment by taranjeet** [comment]
closing this PR as these apps are not present and for App test case is already added. thanks for the contribution.

**Pull request closed without merging**